subprojects/file-collections/src/main/java/org/gradle/api/internal/file/collections/SingleIncludePatternFileTree.java

```diff
@@ -81,10 +82,10 @@ public void visitStructure(MinimalFileTreeStructureVisitor visitor, FileTreeInte
 
     @Override
     public void visit(FileVisitor visitor) {
-        doVisit(visitor, baseDir, new LinkedList<>(), 0, new AtomicBoolean());
+        doVisit(visitor, baseDir, new ArrayDeque<>(), 0, new AtomicBoolean());
     }
 
-    private void doVisit(FileVisitor visitor, File file, LinkedList<String> relativePath, int segmentIndex, AtomicBoolean stopFlag) {
+    private void doVisit(FileVisitor visitor, File file, Deque<String> pathSegments, int segmentIndex, AtomicBoolean stopFlag) {
```

Comment:
🧐 I like the change to `ArrayDeque` but why rename the parameter? It _is_ holding a relative path, no?

Reply:
It is confusing. I expect `relativePath` to have RelativePath type, not some kind of array